pyvrp/cpp/educate/LocalSearch.cpp

```diff
@@ -20,6 +20,12 @@ void LocalSearch::search(Individual &indiv)
     if (nodeOps.empty())
         throw std::logic_error("No known node operators.");
 
+    int neighbourhood_size = 0;
+    for (auto const client : orderNodes)
+        neighbourhood_size += neighbours[client].size();
+    if (neighbourhood_size == 0)
+        throw std::runtime_error("Granular neighbourhood is empty.");
```

Comment:
I'm thinking of what the right approach here is. Do we only want to raise when the entire neighbourhood is empty? It already feels somewhat weird to me if any neighbourhood is empty, for any client. Should we raise in that case? Or are there valid use cases where one might have a zero-size neighbourhood for a particular client?

Reply:
It's a bit subtle, I can imagine situations where e.g. you want to focus the search on a subset of the solution, so then, yes it's valid so let's use the most general option.